### DISCUSSION TITLE
Best way to generate a schema file from what was built programmatically?

### DISCUSSION THREAD

**Discussion by aleksandarsusnjar** [graphql-java/graphql-java#2322] [Q&A] [answered]
Is there a simple, purely `graphql-java` way to generate SDL text/file from what is built programmatically using `GraphQLSchema.Builder`?

If not, what is the next best way?

- I do know that I can use other introspection but I'd prefer not to if I don't have to.
- I'd prefer not to have to write a _ton_ of Java code to do this.

**Comment by deepakguruu** [comment, marked as answer]
Refer #2284

**Reply by aleksandarsusnjar** [reply]
Thanks! That looks awesome.
